http/middleware.go

```diff
@@ -43,6 +44,18 @@ func WrapHandler(waf coraza.WAF, l Logger, h http.Handler) http.Handler {
 		// We continue with the other middlewares by catching the response
 		h.ServeHTTP(wrap(w, r, tx), r)
 
+		for k, vv := range w.Header() {
+			for _, v := range vv {
+				tx.AddResponseHeader(k, v)
+			}
+		}
+		status, _ := strconv.Atoi(w.Header().Get("status"))
```

Comment:
These aren't HTTP headers, `writeHeader(status)` will delegate straight into the response line.

I think we need a wrapper similar to this one

https://github.com/http-wasm/http-wasm-host-go/blob/main/handler/nethttp/buffer.go#L31

For protocol, just use the request protocol (we can probably remove proto parameter from `ProcessResponseHeaders` since response and request protocol should always match)

Reply:
Thanks, took inspiration also from [Zipkin](https://github.com/openzipkin/zipkin-go/blob/master/middleware/http/server.go) and used the interceptor to save the status code inside the middleware

Reply:
Could you clarify what is expected to change here @anuraaga ?